joint_trajectory_controller/test/test_trajectory_controller_utils.hpp

```diff
@@ -391,6 +392,22 @@ class TrajectoryControllerTest : public ::testing::Test
     return traj_controller_->get_node()->activate();
   }
 
+  rclcpp_lifecycle::State DeactivateTrajectoryController()
+  {
+    if (traj_controller_)
+    {
+      if (
+        traj_controller_->get_lifecycle_state().id() ==
+        lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE)
+      {
+        const auto new_state = traj_controller_->get_node()->deactivate();
+        EXPECT_EQ(new_state.id(), lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE);
+        return new_state;
+      }
+    }
+    return traj_controller_->get_lifecycle_state();
+  }
```

Comment:
do we really need to return anything here?

```suggestion
    if (traj_controller_)
    {
      if (
        traj_controller_->get_lifecycle_state().id() ==
        lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE)
      {
        traj_controller_->get_node()->deactivate();
      }
    }
  }
```

Reply:
You are right. I added this before adding that EXPECT_EQ statement

Reply:
Bence can we leave the `EXPECT_EQ` part? or you prefer not to have it?

Reply:
We could leave it, good point!

Reply:
ok

Reply:
```suggestion
    if (traj_controller_)
    {
      if (
        traj_controller_->get_lifecycle_state().id() ==
        lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE)
      {
        const auto new_state = traj_controller_->get_node()->deactivate();
        EXPECT_EQ(new_state.id(), lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE);
      }
    }
  }
```